jormungandr-integration-tests/src/common/jormungandr/starter.rs

```diff
@@ -185,3 +196,63 @@ pub fn assert_start_jormungandr_node_as_passive_fail(
 
     process_assert::assert_process_failed_and_matches_message(command, &expected_msg);
 }
+
+pub fn start_jormungandr_node_as_passive_with_log_verification(
+    config: &JormungandrConfig,
+    timeout_value: u64,
+) -> Result<JormungandrProcess, StartupError> {
+    start_jormungandr_node_as_passive_with_timeout_and_log_checks(
+        &config,
+        timeout_value,
+        |logger: &JormungandrLogger| {
+            logger.message_logged_multiple_times("initial bootstrap completed", 2)
+        },
+        |logger: &JormungandrLogger| logger.contains_error(),
+    )
+}
+
+fn start_jormungandr_node_as_passive_with_timeout_and_log_checks<F: 'static, G: 'static>(
```

Comment:
Looking at the proliferation of `start_jormungandr_node_*` functions, I wonder if this API could be made more composable with a builder-style API dialing in the features you want with methods changing the configuration dynamically:

```rust
impl Starter {
    pub fn new() -> Self {
        // init with default values ...
    }

    pub fn timeout(&mut self, timeout: Duration) -> &mut Self {
        self.timeout = timeout;
        self
    }

    pub fn passive(&mut self) -> &mut Self {
        self.passive = true;
        self
    }

    pub fn start(self) -> Result<JormungandrProcess, StartupError> {
        // ...
    }
}
```

Reply:
It could also be `fn start(&self)` so that you could reuse all or most of the configuration to launch multiple instances.

Reply:
Geat idea. i'll implement it in separate pr.